qadence/types.py

```diff
@@ -48,6 +50,27 @@
 ]  # type: ignore
 
 
+@dataclass(frozen=True)
+class Vectors:
+    tensor: ArrayLike
+
+
+@dataclass(frozen=True)
+class DensityMatrices:
+    density_matrix: ArrayLike
+
+
+@dataclass(frozen=True)
+class Samples:
+    samples: list[Counter]
+
+
+@dataclass(frozen=True)
```

Comment:
what about expectation values? also, this would change the convenient torch-like interface quite a bit since you cant use autograd on outputs of qadence methods anymore

Reply:
I guess expectation values can stay a `Tensor` I guess ? I don't know of any user request to post-process expectation values yet.

Sure but that is the price to pay for having an intermediate `State` data structure. I know this is breaking but hopefully not too much.

Reply:
A probably less disruptive solution would be to subclass `Tensor`. WDYT @dominikandreasseitz ?